### PR TITLE
DB initialization

### PR DESCRIPTION
**Pull request by kunnapool**
Completes ticket #72 .

Script to initialize the database. The script works by parsing CSV files in the `db_initialization` directory - one to initialize Professors + their Preferences, and one to initialize an entire base Schedule object (dynamic courses + static courses configured).

The structure/format of the csv matters as index values are hardcoded. Script parses and creates all the necessary tables, and preserves the relationships b/w these objects.

Run the `init_db.sh` script to initialize the db (more details in the README)

-----
Note the small change to one of the Schedule models + migration file that comes with it. This was needed to conform with the data

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -52,34 +56,51 @@ To run a specific app's test cases, use the command
...
 docker-compose exec web python manage.py migrate
 ```
+
+To populate the database with values, run the bash script
+
+```
+./init_db.sh
+```
+
+The script populates the database and writes to the `data/` directory. The script makes sure the initialization is only one once.
```
Comment (Russell-Waterhouse) [inline README.md:99]: Small Typo in "only one once."

File: schedule/Schedule_models.py [hunk 2]
```diff
@@ -34,13 +34,15 @@ def __str__(self):
 
 '''PRIMARY KEY: id (Django auto)'''
 class A_CourseSection(models.Model):
-    professor = models.JSONField()  #{"id": <int>, "name": AppUser.user.first_name + AppUser.user.last_name}
+    professor = models.JSONField(null=True, blank=True)  #{"id": <int>, "name": AppUser.user.first_name + AppUser.user.last_name}
     capacity = models.PositiveIntegerField(default=0)
     timeSlots = models.ManyToManyField(A_TimeSlot, related_name='courseSections') #to associate multiple TimeSlot objects
 
-    def __str__(self):
+    '''def __str__(self):
```
Comment (Russell-Waterhouse) [inline schedule/Schedule_models.py:41]: Why has this been commented out?

**Review by Russell-Waterhouse — APPROVED** [review]
Awesome work guys, code looks good!

**Comment by kunnapool** [comment]
> Hey just a heads up, I tried to run the script and got this:
> 
> ```
> ----- Schedule data successfully saved to the database. -----
> ----- Professor data successfully saved to the database. -----
> touch: cannot touch '.db_init_done': Permission denied
> ```

Interesting; will 👀 later today

**Comment by rbassot** [comment]
I completely forgot to handle the FE-facing `Course` objects in this script, so I just did that now. It was pretty trivial after having done the Algorithms-side model - just a few extra lines of code.

Let me know if the new functionality works, and does not break anything that was already there.

**Review by benguarasci — APPROVED** [review]
It works for me all tables are populated properly. Just need to resolve conflicts

File: README.md (hunk 1 above)
Comment (kunnapool) [reply README.md]: Fixed

**Pull request merged**